src/aotcompile.cpp

```diff
@@ -1500,7 +1500,13 @@ void jl_dump_native_impl(void *native_code,
         TheTriple.setObjectFormat(Triple::COFF);
     } else if (TheTriple.isOSDarwin()) {
         TheTriple.setObjectFormat(Triple::MachO);
-        TheTriple.setOS(llvm::Triple::MacOSX);
+        SmallString<16> Str;
+        Str += "macosx";
+        if (TheTriple.isAArch64())
+            Str += "11.0.0"; // Update this if MACOSX_VERSION_MIN changes
+        else
+            Str += "10.14.0";
+        TheTriple.setOSName(Str);
```

Comment:
```suggestion
        TheTriple.setOSName("darwin18"); // set -mmacosx-version-min=10.14, like Make.inc
```

Reply:
That is not what clang does

Reply:
```
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin23.0.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple arm64-apple-macosx14.0.0
```

Reply:
https://github.com/llvm/llvm-project/blob/a5f446bc4bb1ac78d6852cc8e251a1229899b783/clang/lib/Driver/ToolChains/Darwin.cpp#L1049-L1062

Reply:
Oh, interesting to see how the `-triple` and `-target` values disagree there. Alright, I guess this seems right then, assuming it still triggers the right codepaths in LLVM.

Reply:
It does generate the LC_BUILD_VERSION command, which makes `ld` happier